proposals/0023-asynchronous-program-execution-and-broadcast.md

```diff
@@ -0,0 +1,212 @@
+---
+simd: '0023'
+title: Asynchronous Program Execution and Broadcast
+authors:
+  - Anatoly Yakovenko
+category: Standard/Meta
+type: Core
+status: Draft
+created: 2024-04-29
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+This feature changes how the ledger is broadcast and executed. It
+separates proposing blocks full of user transactions from blocks
+with votes.  It allows for N concurrent builders of user transaction
+blocks, and it allows for asynchronous execution of user blocks.
+
+## Motivation
+
+1. Single leader for user transactions is a bottleneck.  Leaders
+are in a single spot in the world, while clients are all over the
+world. Latency is therefore based on how close a client is to a
+leader.  Leaders also have to spend a ton of resources on prioritization
+of transactions.
+
+2. Executing programs before voting is a bottleneck. Fork choice
```

Comment:
Yes. Nodes need to be able to compute a snapshot once an epoch.

Reply:
Nodes can't fall behind that much because the overall CU limits are set for synchronous execution.  But with the option of async execution it is much easier to catch up.  Raw ledger processing without dealing with the network is 20-30x times faster.